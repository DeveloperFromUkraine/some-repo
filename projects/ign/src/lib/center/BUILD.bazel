load("@angular//:index.bzl", "ng_module")
load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")

package(default_visibility = ["//visibility:public"])

sass_binary(
    name = "style",
    src = "center.scss",
)

ng_module(
    name = "center-module",
    srcs = [
        "center.module.ts",
        "center.component.ts",
    ],
    assets = [":center.css", "center.html"],
    tsconfig = "//:tsconfig.json",
    deps = [
        "@npm//tslib",
        "@npm//@types",
        "@angular//packages/core",
        "@angular//packages/common",
    ]
)
