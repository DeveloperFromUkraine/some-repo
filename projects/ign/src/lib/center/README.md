## Use
    The center component is used for aligning other components into the center of the parent element.

    ### Implementation
        - Center can be used for any component.
        - Center is best used for components that are best meant to be in the center of the page without any            surrounding components, as other components may end up on the edges of the parent (see css). 
    
    ### CSS
        - Any CSS related changes (lines, highlighting, etc) will be extended across the parent component as the    window is resized. 


## API 
    ### Selector
        * ign-center
    
    ### Export As
        * CenterComponent


## Implementation HTML
    <ign-center>This is centered text.</ign-center>


## Implementation TS
    None