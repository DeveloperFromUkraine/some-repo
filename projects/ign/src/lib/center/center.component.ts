import {AfterViewInit, Component} from '@angular/core';

@Component({
  selector: 'ign-center',
  templateUrl: './center.html',
  styleUrls: ['./center.scss'],
})
export class CenterComponent implements AfterViewInit {
  ngAfterViewInit() {
    console.warn(
      `Deprecation warning: 'ign-center' is being removed in the next release. Support for this component will be dropped soon.`
    );
  }
}
