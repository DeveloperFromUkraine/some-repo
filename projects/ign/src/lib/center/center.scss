:host {
  position: relative;
  flex: 1;
  height: 100%;
  div {
    display: flex;
    flex-direction: column;
    text-align: center;
  }
}
