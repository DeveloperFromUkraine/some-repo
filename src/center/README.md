## API 
    ### Selector
        * ign-center
    
    ### Export As
        * CenterComponent

## Implementation HTML
    <ign-center>This is centered text.</ign-center>

## Implementation TS
    None