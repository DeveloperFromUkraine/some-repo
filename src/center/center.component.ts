/* tslint:disable:max-classes-per-file */
import { Component } from '@angular/core';

@Component({
  selector: 'ign-center',
  templateUrl: './center.html',
  styleUrls: ['./center.scss'],
})
export class CenterComponent {}
