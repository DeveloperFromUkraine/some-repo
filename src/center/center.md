*** HTML ***
# Center

## Use
The center component is used for aligning other components into the center of the parent element.

## Sample

<mat-tab-group>
    <mat-tab label="Component Sample"><div class="tab-height">
        <ign-center>This is centered text.</ign-center>
    </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            &lt;ign-center&gt;This is centered text.&lt;/ign-center&gt; </table></div>
    </mat-tab>
</mat-tab-group>

## Content
* Elements that are centered include buttons and text in dialog components, images, and other components
at the discretion of the developer.


## Style
* The center component is used for centering elements without needing to utilize the center property in css. The css center
property is not needed and should be avoided when utilizing this component.

*** TypeScript *** 

*** ExternalTypeScript ***

*** Imports ***