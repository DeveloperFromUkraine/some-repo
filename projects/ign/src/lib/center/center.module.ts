import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { CenterComponent } from './center.component';

@NgModule({
    imports: [
        CommonModule
    ],
    exports: [
        CenterComponent
    ],
    declarations: [
        CenterComponent
    ]
})
export class CenterModule {

}
