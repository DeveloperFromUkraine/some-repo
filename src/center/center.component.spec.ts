import { CenterComponent } from "center/center.component";
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ComponentTest } from "../../test/test-bed/component";


describe('CenterComponent', () => {
    let fixture: ComponentFixture<CenterComponent>;

    beforeEach(async () => {
        await ComponentTest.createTestBed([], [CenterComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(CenterComponent);
    });

    it('should match snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });
});