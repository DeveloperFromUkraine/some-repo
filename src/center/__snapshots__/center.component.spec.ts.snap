// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterComponent should match snapshot 1`] = `
<ign-center>
  <div>
    
    
    
  
  </div>
  
</ign-center>
`;
